ttnn/cpp/ttnn/operations/transformer/sdpa/device/kernels/dataflow/reader_interleaved.cpp

```diff
@@ -124,7 +161,10 @@ void kernel_main() {
 
                 cb_push_back(cb_q_in, q_chunk_tiles);
 
-                const uint32_t q_low_idx =
+                if constexpr (is_chunked) {
+                    q_chunk = chunked_q_chunk_offset + q_chunk;
```

Comment:
This logic applies to the writer and compute kernels as well.
If chunked, the logical q_chunk is actually offset based on which chunk we are in. This is used for loop bounds over k chunks and for determining where the diagonal is for applying the causal mask.